File: Examples/ObjectiveC/HillshadeExample.m

#import "HillshadeExample.h"
@import Mapbox;

NSString *const MBXExampleHillshade = @"HillshadeExample";
@interface HillshadeExample () <MGLMapViewDelegate>

@end

@implementation HillshadeExample

- (void)viewDidLoad {
    [super viewDidLoad];
    
    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle lightStyleURL]];
    [mapView setCenterCoordinate:CLLocationCoordinate2DMake(63.6304, -19.6067) zoomLevel:8 animated:false];
    mapView.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
    mapView.tintColor = [UIColor darkGrayColor];
    mapView.delegate = self;
    [self.view addSubview:mapView];
}

- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
    NSURL *url = [NSURL URLWithString:@"mapbox://mapbox.terrain-rgb"];
    MGLRasterDEMSource *source = [[MGLRasterDEMSource alloc] initWithIdentifier:@"terrain-rgb" configurationURL:url];
    [style addSource:source];
    
    MGLHillshadeStyleLayer *layer = [[MGLHillshadeStyleLayer alloc] initWithIdentifier:@"terrain-rgb" source:source];
    layer.hillshadeAccentColor = [NSExpression expressionForConstantValue:[UIColor lightGrayColor]];
    layer.hillshadeShadowColor = [NSExpression expressionForConstantValue:[[UIColor darkGrayColor] colorWithAlphaComponent:0.6]];
    layer.hillshadeHighlightColor = [NSExpression expressionForConstantValue:[UIColor lightGrayColor]];
    
    MGLStyleLayer *symbolLayer = [style layerWithIdentifier:@"waterway-river-canal"];
    [style insertLayer:layer aboveLayer:symbolLayer];
    
}
@end

File: Examples/Swift/HillshadeExample.swift
import Mapbox

@objc(HillshadeExample_Swift)
class HillshadeExample: UIViewController, MGLMapViewDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let mapView = MGLMapView(frame: view.bounds, styleURL: MGLStyle.lightStyleURL)
        mapView.setCenter(CLLocationCoordinate2D(latitude: 63.6304, longitude: -19.6067), zoomLevel: 8, animated: false)
        mapView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
        mapView.tintColor = .darkGray
        mapView.delegate = self
        view.addSubview(mapView)
    }
    
    func mapView(_ mapView: MGLMapView, didFinishLoading style: MGLStyle) {
        if let url = URL(string: "mapbox://mapbox.terrain-rgb") {
            let source = MGLRasterDEMSource(identifier: "terrain-rgb", configurationURL: url)
            style.addSource(source)
            
            let layer = MGLHillshadeStyleLayer(identifier: "terrain-rgb", source: source)
            //            layer.hillshadeAccentColor = NSExpression(forConstantValue: UIColor(red:0.03, green:0.44, blue:0.09, alpha:1.0))
            layer.hillshadeAccentColor = NSExpression(forConstantValue: UIColor.lightGray)
            layer.hillshadeShadowColor = NSExpression(forConstantValue: UIColor.darkGray.withAlphaComponent(0.6))
            layer.hillshadeHighlightColor = NSExpression(forConstantValue: UIColor.lightGray.withAlphaComponent(1.0))
            if let symbolLayer = style.layer(withIdentifier: "waterway-river-canal") {
                style.insertLayer(layer, below: symbolLayer)
            } else {
                style.addLayer(layer)
            }
        }
    }
}
